pkg/cache/v2/cache.go

```diff
@@ -51,21 +52,90 @@ type Cache interface {
 	Fetch(context.Context, Request) (*Response, error)
 }
 
+type ResponseIface interface {
+	GetDiscoveryResponse() (*discovery.DiscoveryResponse, error)
+
+	GetRequest() *discovery.DiscoveryRequest
+
+	GetVersion() string
+}
+
 // Response is a pre-serialized xDS response.
 type Response struct {
+	ResponseIface
 	// Request is the original request.
 	Request discovery.DiscoveryRequest
 
 	// Version of the resources as tracked by the cache for the given type.
 	// Proxy responds with this version as an acknowledgement.
 	Version string
 
-	// The value indicating whether the resource is marshaled, and only one of `Resources` and `MarshaledResources` is available.
-	ResourceMarshaled bool
-
 	// Resources to be included in the response.
 	Resources []types.Resource
 
+	// The value indicating whether the resource is marshaled, and only one of `Resources` and `MarshaledResources` is available.
+	isResourceMarshaled bool
+
 	// Marshaled Resources to be included in the response.
-	MarshaledResources []types.MarshaledResource
+	marshaledResponse *discovery.DiscoveryResponse
+}
+
+type PassthroughResponse struct {
+	ResponseIface
+	// Request is the original request.
+	Request discovery.DiscoveryRequest
+
+	// Version of the resources as tracked by the cache for the given type.
+	// Proxy responds with this version as an acknowledgement.
+	Version string
```

Comment:
I'm wondering whether we can drop these two fields since they're already embedded in the `DiscoveryResponse`. Keeping them may just increase the odds of bugs where the values don't match the `DiscoveryResponse` fields.

Reply:
you're right. I'll get rid of these.